### ISSUE TITLE
Bug - Route53 IDs inconsistant with API

### ISSUE THREAD

**Issue by zkarpinski** [getmoto/moto#8153]
I'll try to add more info over next few days.

TLDR: Route53's API specs are inconsistent with the resource IDs. Some return `/hostedzone/abcd1234567` others return `abc1234567`

It appears _most_ apis only accept the true ID and not the `/%resource_type%` prefix. Take for instance `list_tags_for_resources
` if you provide the full id  `/hostedzone/abcd1234567` the resource is not found or the ID parameter greater than the allowed character limit for the aws cli.
This same API's response returns the shorter resource id
`'ResourceTagSets': [{'ResourceType': 'hostedzone', 'ResourceId': 'AVGQ56HKPAQLME0', 'Tags': [{'Key': 'Deploy', 'Value': 'True'}, {'Key': 'Name', 'Value': 'UnitTest'}]}]}
`

A counter example is list_hosted_zones, it return the longer resource id `/hostedzone/abcd1234567`

It seems the route53 implementation may need some refactoring. As a short term fix, perhaps we can accept both "ID" styles and handle appropriately?

@bblommers thoughts?

**Comment by bblommers** [comment]
Well, that's a pain, and definitely weird behaviour!

> As a short term fix, perhaps we can accept both "ID" styles and handle appropriately?

Absolutely, that's probably the best way forward. 

If/when we determine the exact rules when short resource ID is used/accepted, and when the long version, we can always think about a more thorough refactor. But for now - maximum flexibility in what Moto accepts is best IMO.

**Comment by bblommers** [comment]
FYI: I've just raised #8984 which validates the input for `change_tags_for_resource()` and `list_tags_for_resource()`, so that only the short resource ID is accepted.

That PR also includes a framework for running tests against AWS, so that will be useful when determining the correct behaviour for other methods.

**Comment by bpandola** [comment]
Closing this out as the original (specific) issue was addressed in #8984 and the more general issue of when to return short ids vs prefixed ids in responses has been addressed in #9875 (which uses the Botocore model to ensure parity).

If there are other validation issues with respect to input--whether or not an API endpoint accepts the short id, the prefixed id, or both--we will address those going forward as they manifest and are reported.
